### PR TITLE
Add compiler version checks for what we want to support going forward,

### PR DESCRIPTION
**Pull request by morlovich**
and make us actually build with clang (the log_message_handler.cc change
and config.make change)

### CODE REVIEW

**Comment by morlovich** [comment]
@jeffkaufman any chance you could bump this in your queue? It's sort of release-blocking, after all...

File: config [hunk 1]
```diff
@@ -234,4 +251,46 @@ END
...
+ngx_feature_incs=""
+ngx_feature_path=""
+ngx_feature_libs="-lstdc++"
+ngx_feature_test="
+
+#if defined(__clang__) && defined(__GLIBCXX__)
+   // See https://gcc.gnu.org/onlinedocs/libstdc++/manual/abi.html#abi.versioning 
+   // for a list of various values of __GLIBCXX__. Note that they're not monotonic
+   // with respect to version numbers.
+   #if __GLIBCXX__ == 20120322  ||__GLIBCXX__ == 20120614
```
Comment (jeffkaufman) [inline config]: nit: whitespace around the `||`

File: config [hunk 2]
```diff
@@ -234,4 +251,46 @@ END
...
+      #error \"clang is using libstdc++ 4.7.0 or 4.7.1, which can cause binary incompatibility.\"
+   #endif
+#endif
+
+#if !defined(__clang__) && defined(__GNUC__)
+   #if __GNUC__ < 4 || (__GNUC__ == 4 && __GNUC_MINOR__ < 8)
+      #error \"GCC < 4.8 no longer supported. Please use gcc >= 4.8 or clang >= 3.3\"
```
Comment (jeffkaufman) [inline config]: We have instructions for how to do this easily, right?  Can this generate a link to them?

File: config [hunk 3]
```diff
@@ -234,4 +251,46 @@ END
...
+   #endif
+#endif
+
+#if defined(__clang__)
+   #if __clang_major__ < 3 || (__clang_major__ == 3 && __clang_minor__ < 3)
+      #error \"Please use gcc >= 4.8 or clang >= 3.3\"
```
Comment (jeffkaufman) [inline config]: same here

**Comment by jeffkaufman** [comment]
aside from minor comments, LGTM

File: config (hunk 2 above)
Comment (morlovich) [inline config]: See line 289 below. That's what's ultimately printed to the user. This is what goes to autoconf.err

File: config (hunk 1 above)
Comment (morlovich) [inline config]: Good point. Now I just need to remember where I put this. 
Oh, I guess I could just pull this, can't I?

Comment (jeffkaufman) [inline config]: You can just check out the repo with git, check out the morlovich-compiler-req, make your edit, and push back up.

File: config (hunk 2 above)
Comment (jeffkaufman) [inline config]: Sounds good.

**Comment by morlovich** [comment]
Should I merge this, Jeff?

**Comment by jeffkaufman** [comment]
Yes, go ahead and merge.

**Pull request merged**